akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGroupBySpec.scala

```diff
@@ -148,7 +148,7 @@ class FlowGroupBySpec extends StreamSpec {
     }
 
     "accept cancellation of substreams" in assertAllStagesStopped {
-      new SubstreamsSupport(groupCount = 2) {
+      new SubstreamsSupport(groupCount = 2, maxSubstreams = 3) {
```

Comment:
This test now fails without the `maxSubstreams = 3` because the default leads to `maxSubstreams` being `2` which is how many sub streams there will be (It's grouped by `n % groupCount`)? This makes me worry that the change breaks user code.

Reply:
The change could indeed break user code. But then I wonder how many users are also experiencing the unexpected memory retention... :-)

Reply:
What I wonder is if the boundary should be a `+1` before failing so that this test would pass without modification, but perhaps I do not understand the consequence of the change well enough.

Reply:
Adding +1 would just fudge the tests.

The reality is that most users will have a higher maxSubStreams anyhow...

Reply:
Hmm. Ok. I'm still curios why it needs exactly the group count + 1 to not fail. If the test never starts more than 2 substreams, there shouldn't be an additional closed sub stream causing it to fail?

Reply:
@huntc any chance you could verify this? I'm still worried that if it breaks a test with `maxSubstreams` like this, it will break user code. If that is so, can the memory retention be avoided without breaking user code?